tests/unit/amo/components/TestReportUserAbuse.js

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+
+import ReportUserAbuse, {
+  ReportUserAbuseBase,
+} from 'amo/components/ReportUserAbuse';
+import { setError } from 'core/actions/errors';
+import {
+  hideUserAbuseReportUI,
+  loadUserAbuseReport,
+  sendUserAbuseReport,
+  showUserAbuseReportUI,
+} from 'amo/reducers/userAbuseReports';
+import DismissibleTextForm from 'ui/components/DismissibleTextForm';
+import ErrorList from 'ui/components/ErrorList';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
+import {
+  createFakeUserAbuseReport,
+  createStubErrorHandler,
+  createUserAccountResponse,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  const defaultRenderProps = {
+    errorHandler: createStubErrorHandler(),
+    i18n: fakeI18n(),
+    store: dispatchClientMetadata().store,
+    user: createUserAccountResponse({ id: 9 }),
+  };
+
+  function renderShallow({ ...props } = {}) {
+    return shallowUntilTarget(
+      <ReportUserAbuse {...defaultRenderProps} {...props} />,
+      ReportUserAbuseBase
+    );
+  }
+
+  it('renders a disabled button if no user exists', () => {
+    const root = renderShallow({ user: null });
+
+    expect(root.find('.ReportUserAbuse')).toHaveLength(1);
+    expect(root.find('.ReportUserAbuse-show-more'))
+      .toHaveProp('disabled', true);
+  });
+
+  it('renders a button to report an user', () => {
+    const root = renderShallow();
+
+    expect(root.find('.ReportUserAbuse')).toHaveLength(1);
+    expect(root.find('.ReportUserAbuse-show-more').prop('children'))
+      .toEqual('Report this user for abuse');
+  });
+
+  it('shows the preview content when first rendered', () => {
+    const root = renderShallow();
+
+    expect(root.find('.ReportUserAbuse--is-expanded')).toHaveLength(0);
+  });
+
+  it('shows more content when showReportUI is called', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const user = createUserAccountResponse();
+    const root = renderShallow({ store, user });
+
+    root.instance().showReportUI();
+
+    sinon.assert.calledWith(dispatchSpy, showUserAbuseReportUI({ user }));
+  });
+
+  it('hides more content when hideReportUI is called', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const user = createUserAccountResponse();
+    const root = renderShallow({ store, user });
+
+    root.instance().showReportUI();
+    dispatchSpy.reset();
+
+    root.instance().hideReportUI();
```

Comment:
For these kind of callbacks I think it's better to simulate what `DismissibleTextForm` will do. This will enforce the contract that `ReportUserAbuse` has with the interface of `DismissibleTextForm`. Example:

```js
const form = root.find(DismissibleTextForm);
const dismiss = form.prop('onDismiss');
// Simulate a dismissal:
dismiss();
```

If you end up doing this a lot you can make a helper for it like `simulateDimissal(root)`

Reply:
Ah, that's aces.